paddle/fluid/operators/reshape_op.cc

```diff
@@ -216,7 +216,7 @@ class ReshapeKernel {
     if (shape_tensor) {
       auto *shape_data = shape_tensor->data<int>();
       framework::Tensor cpu_shape_tensor;
-      if (platform::is_gpu_place(ctx.GetPlace())) {
+      if (platform::is_gpu_place(shape_tensor->place())) {
```

Comment:
We should use `shape_tensor->place()` to determine whether to copy `shape_tensor` or not.